src/components/Composer/index.js

```diff
@@ -197,46 +207,73 @@ class Composer extends React.Component {
     }
 
     /**
-     * Handles all types of drag-N-drop events on the composer
+     * DragEvent
      *
      * @param {Object} e native Event
      */
-    dragNDropListener(e) {
-        let isOriginComposer = false;
-        const handler = () => {
-            // Setting dropEffect for dragover is required for '+' icon on certain platforms/browsers (eg. Safari)
-            switch (e.type) {
-                case 'dragover':
-                    e.preventDefault();
-                    e.dataTransfer.dropEffect = COPY_DROP_EFFECT;
-                    this.props.onDragOver(e, isOriginComposer);
-                    break;
-                case 'dragenter':
+    dropZoneDragHandler = (e) => {
+        // Setting dropEffect for dragover is required for '+' icon on certain platforms/browsers (eg. Safari)
+        switch (e.type) {
+            case 'dragover':
+                // Continuous event -> can hurt performance, be careful when subscribing
+                e.dataTransfer.dropEffect = COPY_DROP_EFFECT;
+                this.dropZoneDragState = 'dragover';
+                this.props.onDragOver(e);
+                break;
+            case 'dragenter':
+                // Avoid reporting onDragEnter for children views -> not performant
+                if (this.dropZoneDragState === 'dragleave') {
                     e.dataTransfer.dropEffect = COPY_DROP_EFFECT;
-                    this.props.onDragEnter(e, isOriginComposer);
-                    break;
-                case 'dragleave':
-                    this.props.onDragLeave(e, isOriginComposer);
-                    break;
-                case 'drop':
-                    this.props.onDrop(e, isOriginComposer);
-                    break;
-                default: break;
-            }
-        };
+                    this.dropZoneDragState = 'dragenter';
+                    this.props.onDragEnter(e);
+                }
+                break;
+            case 'drop':
+                this.dropZoneDragState = 'drop';
+                this.props.onDrop(e);
+                break;
+            default: break;
+        }
+    };
 
-        // We first check if drop target is composer so that it can be highlighted
-        if (this.textInput.contains(e.target)) {
-            isOriginComposer = true;
-            handler();
+    /**
+     * Handles all types of drag-N-drop events on the drop zone associated with composer
+     *
+     * @param {Object} e native Event
+     */
+    dropZoneDragListener(e) {
+        if (!this.dropZone) {
```

Comment:
Is there a possibility where dropZone will not be present? If no, please remove this check.